dvc/remote/gdrive.py

```diff
@@ -41,30 +42,18 @@ def __init__(self, path):
         )
 
 
-@decorator
-def _wrap_pydrive_retriable(call):
+def gdrive_retry(func):
     from pydrive2.files import ApiRequestError
 
-    try:
-        result = call()
-    except ApiRequestError as exception:
-        retry_codes = ["403", "500", "502", "503", "504"]
-        if any(
-            "HttpError {}".format(code) in str(exception)
-            for code in retry_codes
-        ):
-            raise GDriveRetriableError("Google API request failed")
-        raise
-    return result
-
+    retry_re = re.compile(r"HttpError (403|500|502|503|504)")
 
-gdrive_retry = compose(
     # 15 tries, start at 0.5s, multiply by golden ratio, cap at 20s
-    retry(
-        15, GDriveRetriableError, timeout=lambda a: min(0.5 * 1.618 ** a, 20)
-    ),
-    _wrap_pydrive_retriable,
-)
+    return retry(
```

Comment:
is there a simple way to write a debug message on a retry, btw?

Reply:
You can wrap a func, which you retry into [`@log_errors()`](https://funcy.readthedocs.io/en/stable/debug.html#log_errors):
```python
# ...
return retry(...)(log_errors(logger.error, stack=False)(func))
```
This will show errors, but there is no non-hacky way to show retries.